packages/domain/lib/usecase/get_movie_response_usecase.dart

```diff
@@ -0,0 +1,60 @@
+import '../enum/movie_type.dart';
+import '../utils/const.dart';
+import '../model/movie_response_model.dart';
+import '../repository/movie_repository.dart';
+import 'usecase.dart';
+
+class GetMovieResponseUseCase
+    implements UseCaseParams<MovieType, Future<List<MovieResponse>>> {
+  final NetworkRepository _repository;
+
+  GetMovieResponseUseCase(this._repository);
+
+  @override
+  Future<List<MovieResponse>> call(movieType) async {
+    final List<MovieResponse> jsonMovies = [];
+    final response = (movieType == MovieType.anticipated)
+        ? await _repository.getMovieAnticipatedData(
+            queryParameters: {
+              'extended': 'full',
+              'page': 1,
+            },
+          )
+        : await _repository.getMovieTrendingData(
+            queryParameters: {
+              'extended': 'full',
+              'page': 1,
+            },
+          );
+    final pageCount = int.parse(response.headers[Config.pageCount][0]);
```

Comment:
can we avoid dynamic in headers?
if headers is a map -> we get value by key -> then why we need [0]?
response.headers[Config.pageCount] will looks good

Reply:
bc it is map of <String, List<String>>